src/kubernetes/stream.rs

```diff
@@ -35,6 +35,8 @@ where
             }
         }
         decoder.finish().map_err(|data| Error::UnparsedDataUponCompletion { data })?;
+
+        ()
```

Comment:
Why is this needed?

Reply:
It's to satisfy clippy, though I'm not entirely sure why it is needed or if there is a better way to resolve that lint.

https://github.com/timberio/vector/pull/4818/files/471caf28164db3599611494d68efbcb0ab090ae6#diff-38f65b460ca0ce88f0e24ff9791321090396872183b32b03c8a3435dbe4f45f3

We could add a comment here about it though. I could see it being confusing.

Reply:
Hm, strange. I'm fine to leave it and hopefully a later clippy update will tell us it's unnecessary :smile: